specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json

```diff
@@ -752,16 +752,32 @@
           "description": "Minimal TLS version. Allowed values: 'None', '1.0', '1.1', '1.2'",
           "type": "string"
         },
-        "storageAccountType": {
-          "description": "The storage account type used to store backups for this instance. The options are LRS (LocallyRedundantStorage), ZRS (ZoneRedundantStorage) and GRS (GeoRedundantStorage)",
+        "currentBackupStorageRedundancy": {
+          "description": "The storage account type used to store backups for this instance. The options are Local (LocallyRedundantStorage), Zone (ZoneRedundantStorage), Geo (GeoRedundantStorage) and GeoZone(GeoZoneRedundantStorage)",
           "enum": [
-            "GRS",
-            "LRS",
-            "ZRS"
+            "Geo",
+            "Local",
+            "Zone",
+            "GeoZone"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CurrentBackupStorageRedundancy",
```

Comment:
Could there just be one Definition of BackupStorageRedundancy enum type, which is referenced for both currentBackupStorageRedundancy and requestedBackupStorageRedundancy properties? So that the values are obviously interchangeable? Or are they likely to diverge in the future?

Reply:
We have to have both currentBackupStorageRedundancy and requestedBackupStorageRedundancy because we are soon going to enable backup residency change feature. That way  currentBackupStorageRedundancy and requestedBackupStorageRedundancy will not point to the same value while the change is in progress. For Managed Instance, this is a size of data operation and can last up to 24h. Also the parameters are aligned with sql database parameters now and that was a decision from all of our PMs

Reply:
@svmicano Let me try to clarify. I'm not saying that they point to the same value. I'm saying that it looks like they point to the same *type* of value, so you shouldn't really need to declare two separate enum types.

Reply:
They do point to the same type of value but I was not aware that there is a better way to do it. I aligned the api with SQLDB design which is like this. Since these APIs are chasing Ignite, is it OK to create a follow up PR for this when we get time?

Reply:
@svmicano  In that case it could become a small, but SDK-breaking change to unify the types in your next PR, so I'd recommend you instead fix it in this PR. By creating a Definition to $ref.

Reply:
@TimLovellSmith 
it seems that we need to let PM or owners sync to fix it in the next API version. 
We are targeting to release this version as soon as possible to unblock downstream client development for Ignite conference.

Reply:
@ericshape I still don't understand why you would postpone this one. I think you have time to fix it; it should not be hard to update the swagger in the PR to fix.

Reply:
addressed and update the enum name